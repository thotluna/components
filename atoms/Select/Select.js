import React from 'react'
import PropTypes from 'prop-types'
import styles from './Select.module.css'
import { getClasses } from '../../helpers/styles'

export const Select = ({
  id,
  isInline,
  isHideLabel,
  label,
  onChange,
  options,
  ref,
  value,
}) => {
  const getStyles = getClasses(styles)({})
  return (
    <label className={getStyles('label')}>
      <span
        className={getStyles('span', {
          'is-hide': isHideLabel,
        })}
      >
        {label}
      </span>
      <select
        className={getStyles('container', {
          'is-inline': isInline,
        })}
        onChange={(event) => onChange(event?.currentTarget?.value)}
        value={value}
        id={id}
        aria-label={label}
        ref={ref}
      >
        <option key="0" value={null}></option>
        {options.map(({ text, value }) => (
          <option key={value} value={value}>
            {text}
          </option>
        ))}
      </select>
    </label>
  )
}

Select.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    })
  ).isRequired,
  isInline: PropTypes.bool,
  isHideLabel: PropTypes.bool,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  ref: PropTypes.any,
  onChange: PropTypes.func,
}

Select.defaultProps = {
  isHideLabel: true,
}

export default Select
