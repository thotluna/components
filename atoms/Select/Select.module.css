.container {
  display: block;
  width: 100%;
  max-width: 100%;
  box-sizing: border-box;
  padding: 12px 30px 12px 17px;
  border: var(--border-width-thin) solid var(--color-primary);
  margin: 0;
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
  background-color: var(--input-background);
  background-image: url('/image/icon.svg');
  background-position: right 0.7em top 50%, 0 0;
  background-repeat: no-repeat, repeat;
  background-size: 1em auto, 100%;
  border-radius: var(--input-border-radius);
  color: var(--color-font-base);
  font-family: var(--font-family-sans);
  font-size: 16px;
  font-weight: 700;
  line-height: var(--line-height-tight);
}

.container::-ms-expand {
  display: none;
}

.container:hover {
  border-color: var(--color-primary);
  box-shadow: var(--box-shadow-focus-in), var(--box-shadow-focus-out);
}

.container:focus {
  box-shadow: var(--box-shadow-focus-in), var(--box-shadow-focus-out);
  outline: none;
}

.is-inline {
  max-width: max-content;
}

.container option {
  font-weight: normal;
}
