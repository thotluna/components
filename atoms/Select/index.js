export { default } from './Select'
export { default as styles } from './Select.module.css'
