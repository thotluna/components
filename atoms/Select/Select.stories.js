import React from 'react'
import Select from './index'

const Template = (args) => <Select {...args} />

export default {
  title: 'Atoms/Select',
  component: Select,
  args: {
    options: [
      { text: 'First', value: 1 },
      { text: 'Second', value: 2 },
      { text: 'Third', value: 3 },
    ],
    value: 2,
  },
  argTypes: {
    options: {
      description: '**array of shaped objects:**',
      table: {
        type: {
          summary: 'object',
          detail: "{ text: 'string', value: 'string|number'}",
        },
      },
    },
    onChange: { table: { disable: true } },
    ref: { table: { disable: true } },
  },
}

const args = {
  isInline: false,
  isHideLabel: true,
  label: 'Select',
  options: [
    { text: 'First', value: 1 },
    { text: 'Second', value: 2 },
    { text: 'Third', value: 3 },
  ],
  value: 2,
}

export const Default = Template.bind({})
Default.args = { id: '#1', ...args }

export const Value = Template.bind({})
Value.args = { id: '#2', ...args, value: 2 }

export const Inline = Template.bind({})
Inline.args = { id: '#3', ...args, isInline: true }

export const LabelShow = Template.bind({})
LabelShow.args = { id: '#4', ...args, isHideLabel: false }
