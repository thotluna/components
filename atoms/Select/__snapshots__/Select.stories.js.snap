// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Select Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <label
    className="label"
  >
    <span
      className="span is-hide"
    >
      Select
    </span>
    <select
      aria-label="Select"
      className="container"
      id="#1"
      onChange={[Function]}
      value={2}
    >
      <option
        value={null}
      />
      <option
        value={1}
      >
        First
      </option>
      <option
        value={2}
      >
        Second
      </option>
      <option
        value={3}
      >
        Third
      </option>
    </select>
  </label>
</div>
`;

exports[`Storyshots Atoms/Select Inline 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <label
    className="label"
  >
    <span
      className="span is-hide"
    >
      Select
    </span>
    <select
      aria-label="Select"
      className="container is-inline"
      id="#3"
      onChange={[Function]}
      value={2}
    >
      <option
        value={null}
      />
      <option
        value={1}
      >
        First
      </option>
      <option
        value={2}
      >
        Second
      </option>
      <option
        value={3}
      >
        Third
      </option>
    </select>
  </label>
</div>
`;

exports[`Storyshots Atoms/Select Label Show 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <label
    className="label"
  >
    <span
      className="span"
    >
      Select
    </span>
    <select
      aria-label="Select"
      className="container"
      id="#4"
      onChange={[Function]}
      value={2}
    >
      <option
        value={null}
      />
      <option
        value={1}
      >
        First
      </option>
      <option
        value={2}
      >
        Second
      </option>
      <option
        value={3}
      >
        Third
      </option>
    </select>
  </label>
</div>
`;

exports[`Storyshots Atoms/Select Value 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <label
    className="label"
  >
    <span
      className="span is-hide"
    >
      Select
    </span>
    <select
      aria-label="Select"
      className="container"
      id="#2"
      onChange={[Function]}
      value={2}
    >
      <option
        value={null}
      />
      <option
        value={1}
      >
        First
      </option>
      <option
        value={2}
      >
        Second
      </option>
      <option
        value={3}
      >
        Third
      </option>
    </select>
  </label>
</div>
`;
